### PR TITLE
fix phpdoc for ModelToIdPropertyTransformer::$property + add test

### PR DESCRIPTION
**Pull request by dmaicher**
## Subject

I am targeting this branch, because its BC.

Related to https://github.com/sonata-project/SonataAdminBundle/issues/6779 (fix needs to be merged into master first).

## Changelog

```markdown
### Fixed
- fixed phpdoc for `Sonata\AdminBundle\Form\DataTransformer\ModelToIdPropertyTransformer::$property` which can be `string[]` as well
```

### CODE REVIEW

File: src/Form/DataTransformer/ModelToIdPropertyTransformer.php [hunk 1]
```diff
@@ -59,10 +59,10 @@ class ModelToIdPropertyTransformer implements DataTransformerInterface
...
-     * @param string        $property
-     * @param bool          $multiple
-     * @param callable|null $toStringCallback
+     * @param string          $className
+     * @param string|string[] $property
+     * @param bool            $multiple
+     * @param callable|null   $toStringCallback
      *
      * @phpstan-param class-string<T> $className
      * @phpstan-param null|callable(object, string): string $toStringCallback
```
Comment (VincentLanglet) [inline src/Form/DataTransformer/ModelToIdPropertyTransformer.php:68]: Now the callable can accept `string[]` no ?

SO it should be
```
@phpstan-template P of string|string[]
@phpstan-param class-string<T> $className
@phpstan-param null|callable(object, P): string $toStringCallback
```

but I'm not sure

Comment (dmaicher) [reply src/Form/DataTransformer/ModelToIdPropertyTransformer.php]:
```
 PHPDoc tag @template P for method                                      
         Sonata\AdminBundle\Form\DataTransformer\ModelToIdPropertyTransformer:  
         :__construct() with bound type array<string>|string is not supported.

**Comment by VincentLanglet** [comment]
`@phpstan-template P of string|string[]` is not supported, so maybe `@phpstan-template P` ?

File: src/Form/DataTransformer/ModelToIdPropertyTransformer.php (hunk 1 above)
Comment (dmaicher) [reply src/Form/DataTransformer/ModelToIdPropertyTransformer.php]: @VincentLanglet this seems to work?

`@phpstan-param null|callable(object, string|string[]): string $toStringCallback`

Comment (VincentLanglet) [reply src/Form/DataTransformer/ModelToIdPropertyTransformer.php]: did you tried @phpstan-template P ?

Comment (dmaicher) [reply src/Form/DataTransformer/ModelToIdPropertyTransformer.php]: What would be the benefit of that? What is `P` then?

Comment (VincentLanglet) [reply src/Form/DataTransformer/ModelToIdPropertyTransformer.php]: If you write `callable(object, string|string[])`

passing a `callable(object, string)` would be invalid.
but it should be valid if the property is a string.

Comment (dmaicher) [reply src/Form/DataTransformer/ModelToIdPropertyTransformer.php]: But we don't define `P` anywhere? So how does this help?

Anyway I changed it.

**Comment by VincentLanglet** [comment]
Rebasing 3.x should fix the build

**Review by VincentLanglet — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]
Thank you David!

**Comment by franmomu** [comment]
Just an idea, since `ModelToIdPropertyTransformer` apparently is only used in `ModelAutocompleteType` (and the user usually does not interact with `ModelToIdPropertyTransformer`), what about to only allow `array` in `ModelToIdPropertyTransformer` and apply a normalizer to `ModelAutocompleteType` option (to `OptionsResolver::setNormalizer`)  in case is just a string, change it to `array`.

So this way, the user can use `string|string[]` for `property` and the `ModelAutocompleteType` is the one in charge of normalize it to `string[]`.

**Pull request merged**